lib/utils/utils.js

```diff
@@ -64,8 +67,49 @@ function setConsoleLogPrefix(prefix) {
     };
 }
 
+// This should be roughly similar to ARG_MAX, some systems is 131072 per argument 
+// but we're going to be very conservative and cap the total size to be safe.
+const CMD_SIZE_LIMIT = 8000;
+
+/**
+ * Only allows a specific number of bytes for variable names and values.  Any variable exceeding that will be stored in a file
+ * and the variable value will contain the temp file location.
+ * If this substitution occurs then an additional variable called FILE_PARAMS will contain the list of substituted variable names.
+ * @param {Object} vars 
+ * @param {Number} sizeLimit Size limit to uphold
+ * @returns Object of values from the provided one, modified as described.
+ */
+function limitVariableSizes(vars, sizeLimit = CMD_SIZE_LIMIT) {
+    // First sort variables by serialized (string representation) size so we get the smallest variables out first
+    const keys = Object.keys(vars).sort((a,b) => 
+        String(vars[a]).length - String(vars[b]).length
+    );
+    const sizeLimitedVars = {};
+    let totalSize = 0;
+    for (const key of keys) {
+        const newSize = totalSize + String(key).length + String(vars[key]).length;
+        if (newSize >= sizeLimit) {
+            // We have hit our limit and all variables must be stored in temp files now
+            const tmpFile = tmp.fileSync();
+            fs.writeSync(tmpFile.fd, vars[key]);
+            fs.closeSync(tmpFile.fd);
```

Comment:
Idea: Maybe consider adding more smarts in the loop, to close the file descriptor only once after the loop, if it was opened/written to?

Reply:
Well no I want to close the file and make sure there's no open descriptor.  The fact it's a temp file should be dealt with afterwards by the library.  And otherwise I don't want extra file descriptors laying around unnecessarily.  I get the temp file, write to it, close it and then record the path and move on.  I don't think there's any change needed here.

Reply:
Anyway it seems that closing the file is not necessary if I use writeFileSync anyway...

Reply:
Anyway it doesn't seem that closing the file is necessary when using writeFileSync...

Reply:
Then drop it alltogether?

Reply:
Yes ma'am. :)